packages/libs/eda/src/lib/core/components/visualizations/implementations/VolcanoPlotVisualization.tsx

```diff
@@ -148,6 +159,146 @@ function VolcanoPlotViz(props: VisualizationProps<Options>) {
     ])
   );
 
+  /**
+   * Find mins and maxes of the data and for the plot.
+   * The standard x axis is the log2 fold change. The standard
+   * y axis is -log10 raw p value.
+   */
+
+  // Find maxes and mins of the data itself
+  const rawDataMinMaxValues: RawDataMinMaxValues = useMemo(() => {
+    if (!data.value)
+      return {
+        x: { min: 0, max: 0 },
+        y: { min: 0, max: 0 },
+      };
+    const dataXMin = min(data.value.map((d) => Number(d.log2foldChange))) ?? 0;
+    const dataXMax = max(data.value.map((d) => Number(d.log2foldChange))) ?? 0;
+    const dataYMin = min(data.value.map((d) => Number(d.pValue))) ?? 0;
+    const dataYMax = max(data.value.map((d) => Number(d.pValue))) ?? 0;
+    return {
+      x: { min: dataXMin, max: dataXMax },
+      y: { min: dataYMin, max: dataYMax },
+    };
+  }, [data.value]);
+
+  // Determine mins, maxes of axes in the plot. These are different than the data mins/maxes because
+  // of the log transform and the little bit of padding, or because axis ranges are supplied.
+  // NOTE: this state may be unnecessary depending on how we implement user-controlled axis ranges
+  const [xAxisRange, setXAxisRange] =
+    useState<NumberRange | undefined>(undefined);
+  const independentAxisRange = useMemo(() => {
+    if (!data.value) return undefined;
+    if (xAxisRange) {
+      return xAxisRange;
+    } else {
+      const {
+        x: { min: dataXMin, max: dataXMax },
+      } = rawDataMinMaxValues;
+      if (dataXMin && dataXMax) {
+        // We can use the dataMin and dataMax here because we don't have a further transform
+        // Add a little padding to prevent clipping the glyph representing the extreme points
+        return {
+          min: dataXMin - (dataXMax - dataXMin) * AXIS_PADDING_FACTOR,
+          max: dataXMax + (dataXMax - dataXMin) * AXIS_PADDING_FACTOR,
+        };
+      } else {
+        return { min: 0, max: 0 };
+      }
+    }
+  }, [data.value, xAxisRange, rawDataMinMaxValues]);
+
+  // NOTE: this state may be unnecessary depending on how we implement user-controlled axis ranges
+  const [yAxisRange, setYAxisRange] =
```

Comment:
why might this state be unnecessary?

Reply:
Unclear to me if we could just update and use the `vizConfig` data. I think that will become more clear in the follow up tasks to add axis controls and an opacity slider. 🤷